### DISCUSSION TITLE
Camera settings

### DISCUSSION THREAD

**Discussion by persius77** [lllyasviel/Fooocus#2738] [Q&A] [answered]
I'm starting to finally understand how Fooocus works based on tutorials and tips from users.   My latest challenge is with camera settings   I've followed numerous examples but so far, nothing I do or any prompt that I use allows me to adjust my camera setting.    

I am practicing on a simple rendering which is a cabin surrounded by wilderness.  Approx. half of my scenes has cabin roof is cut off and it's always a close up shot.  I've attached an example.
I can't understand how to move the cabin further away from the camera.  I have tried all the prompts that are suggested and nothing is working for me. I have also tried using all the various aspect ratios but again, nothing changes.   

Can somebody please offer a suggestion or two?

Thank you.
![screenshot](https://github.com/lllyasviel/Fooocus/assets/165508031/13008be6-f57b-48ba-a31a-18f0aeb640d0)

**Comment by ILikeToasters** [comment, marked as answer]
So you want shots that are further away? Have you used weights to some parts of the prompt like (extra wide angle:1.5)? You are always going to have some that won't be what you want of course. Add things that might cause it to pull further back like clouds in the sky. Without more details on the prompt to try out that would be were I would start. Some of these might also help if you have not tried: https://ai.kleebz.com/shot-types-and-camera-angles-for-ai-image-generation/

**Reply by persius77** [reply]
Hi.  I did watch the video you linked and it's helpful with characters that I created, however, it's still a little fussy with objects like houses.   Regarding the issue of weights... what is the min and max when it comes to weights?    I know this may seem like a basic question for you but I'm all about taking notes and applying it but I helps when I understand the limits of how far to take weights.  For example, some Loras have a slim margin for weights (based on what I read). If you go over even by .2 or .3., it kills the effect.

**Reply by ILikeToasters** [reply]
I find in prompts, over 1.5 will usually give weird results. Not bad always but weird.

**Comment by Rogala** [comment]
To control the "camera" you need to explain to the AI how far away from you you need to move the focus object.
In our world, we approach or move away from the object, or take different lenses on the camera.

In the real world, if we need to shoot a wide angle, we take a camera with a short focal length. Here it's just aspect ratio, the wider the wider the inherently shorter the focal length. If you need to shoot full-length it's the reverse of the aspect ratio of narrower but taller.

And long focal length for shooting distant objects in the real world is just like a 1024*1024 close-up for the AI.
We need to learn how to move the main object or character away from the edge of the screen, at the desired distance, I have not yet figured out how to do it myself.

We need to learn how to move the world in the picture.

These are real world examples
[Focal Length Guide for Shooting in Large Spaces](https://www.lensrentals.com/blog/2016/03/focal-length-guide-for-shooting-in-large-spaces/)
[Understanding Focal Length](https://www.nikonusa.com/en/learn-and-explore/a/tips-and-techniques/understanding-focal-length.html)

**Reply by persius77** [reply]
Thanks for the links.   I will give these a try shortly.  One question that popped up is, do I use, "  (    )  " when using values like,  (50mm), or,  (F/11), or does that matter?    I read an article for Seaart.ai that that parentheses are required for such values.

**Reply by Rogala** [reply]
I tried both (FL: 18mm) and FL: 18mm, the program does not understand it, there are wildcards where different perspectives are collected, some of them work, but it is better to also screen them with a weight indicator (aerial perspective shot: N)

[Camera Perspectives](https://civitai.com/models/286127/camera-perspectives-507)
[perspective.txt](https://github.com/lllyasviel/Fooocus/files/14929640/perspective.txt)

**Reply by persius77** [reply]
I found your link with the wildcards.   Where is the "extensions" directory so this script can be added?

**Comment by poor7** [comment]
Try this LoRA.
https://huggingface.co/e-n-v-y/envy-zoom-slider-xl-01/resolve/main/EnvyZoomSliderXL01.safetensors
